### PR TITLE
AT-28-IA chat connection

### PR DESCRIPTION
**Pull request by CorreaJose13**
this PR add a feature to use gemini AI through the bot

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,29 +49,32 @@ terraform init
 
 ### 4. Set Sensitive Values
 
-To protect the project sensitive values (e.g., Telegram bot API token and MongoDB token) you can choose one of two methods:
+To protect the project sensitive values (e.g., Telegram bot API token , MongoDB token, etc.) you can choose one of two methods:
```
Comment (johanflorez219) [inline README.md:52]: token, instead token ,

File: README.md [hunk 2]
```diff
@@ -49,29 +49,32 @@ terraform init
...
 
 a. Using a <code>.tfvars</code> file
 
-- Create a file named <code>secrets.tfvars</code> in the lambda directory:
+- Create a file with the extension <code>.tfvars</code> in the lambda directory:
```
Comment (johanflorez219) [inline README.md:56]: with extension *

File: gemini/gemini.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+func getAPIKey() (string, error) {
+	key := os.Getenv("GEM_API_KEY")
+	if key == "" {
+		return "", ErrNoGemApiKey
+	}
+	return key, nil
+}
+
+func extractFirstCandidate(candidates []*genai.Candidate) (string, error) {
+	for _, candidate := range candidates {
```
Comment (johanflorez219) [inline gemini/gemini.go:28]: maybe 
if len(candidates) != 0 && content := candidates[0]; content != nil && len(content.Parts) != nil {
return fmt.Sprintf("%v", content.Parts[0]), nil
}

File: gemini/gemini.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+			continue
+		}
+
+		return fmt.Sprintf("%v", content.Parts[0]), nil
+	}
+
+	return "", ErrNoCandidates
+}
+
+func Gemini(prompt string) (string, error) {
```
Comment (johanflorez219) [inline gemini/gemini.go:40]: improve function name, maybe QueryGemini

File: telegram/commands/gemini/gemini.go [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+var (
+	ErrNoPrompt = errors.New("En serio metiste un prompt vacío bobo hijueputa? Haz algo bien sub-humano 🙄")
+)
+
+func HandleGemini(bot *shared.Bot, chatID int64, userName, content string) error {
+	if content == "" {
+		return ErrNoPrompt
+	}
+	response, err := gemini.Gemini(content + "responde solo texto")
```
Comment (johanflorez219) [inline telegram/commands/gemini/gemini.go:18]: add a comment here, its too hacky and maybe we will not understand what it does if we need to change this code in the future

File: gemini/gemini.go [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+	ErrNoCandidates = errors.New("no candidates found")
+	ErrNoResponse   = errors.New("no response found")
+	ErrNoGemApiKey  = errors.New("missing API key in environment variables")
+)
+
+func getAPIKey() (string, error) {
+	key := os.Getenv("GEM_API_KEY")
+	if key == "" {
+		return "", ErrNoGemApiKey
+	}
```
Comment (bescobarm) [inline gemini/gemini.go:23]: add a space bellow the {

File: gemini/gemini.go (hunk 3 above)
Comment (bescobarm) [reply gemini/gemini.go:28]: this worsens the readability

File: telegram/commands/gemini/gemini.go [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+)
+
+var (
+	ErrNoPrompt = errors.New("En serio metiste un prompt vacío bobo hijueputa? Haz algo bien sub-humano 🙄")
+)
+
+func HandleGemini(bot *shared.Bot, chatID int64, userName, content string) error {
+	if content == "" {
+		return ErrNoPrompt
+	}
```
Comment (bescobarm) [inline telegram/commands/gemini/gemini.go:17]: add a space bellow the {

File: telegram/commands/gemini/gemini.go (hunk 5 above)
Comment (bescobarm) [reply telegram/commands/gemini/gemini.go:18]: additionally, set the string as a global var

File: gemini/gemini.go (hunk 6 above)
Comment (CorreaJose13) [reply gemini/gemini.go:23]: not gonna do it

File: telegram/commands/gemini/gemini.go (hunk 7 above)
Comment (CorreaJose13) [reply telegram/commands/gemini/gemini.go:17]: not gonna do it

**Review by bescobarm — CHANGES_REQUESTED** [review]
spaces after { are needed to merge this pr :v

File: telegram/commands/gemini/gemini.go [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+import (
+	"activity-tracker/gemini"
+	"activity-tracker/shared"
+	"errors"
+	"fmt"
+)
+
+var (
+	errNoPrompt = errors.New("En serio metiste un prompt vacío bobo hijueputa? Haz algo bien sub-humano 🙄")
+	ForceText   = "responde solo texto"
```
Comment (bescobarm) [inline telegram/commands/gemini/gemini.go:12]: why is this var exportable?

Comment (CorreaJose13) [reply telegram/commands/gemini/gemini.go:12]: look dicksucker, u just made a comment above and i quote 'additionally, set the string as a global var' and i did this, if this is not how u do it then i dont wtf to do

File: README.md (hunk 1 above)
Comment (CorreaJose13) [reply README.md]: solved

File: gemini/gemini.go (hunk 3 above)
Comment (CorreaJose13) [reply gemini/gemini.go:28]: decidanse ps

File: gemini/gemini.go (hunk 4 above)
Comment (CorreaJose13) [reply gemini/gemini.go]: solved

File: gemini/gemini.go (hunk 6 above)
Comment (CorreaJose13) [reply gemini/gemini.go:23]: solved

File: telegram/commands/gemini/gemini.go (hunk 8 above)
Comment (bescobarm) [reply telegram/commands/gemini/gemini.go:12]: you are using unnecessary exportable declaration, why do u create something that u won't use?

File: README.md (hunk 2 above)
Comment (CorreaJose13) [reply README.md]: solved

File: telegram/commands/gemini/gemini.go (hunk 7 above)
Comment (CorreaJose13) [reply telegram/commands/gemini/gemini.go:18]: solved

File: telegram/commands/gemini/gemini.go (hunk 8 above)
Comment (CorreaJose13) [reply telegram/commands/gemini/gemini.go:12]: solved, now stop pissing me off or i will touch myself with your profile pic

File: gemini/gemini.go (hunk 6 above)
Comment (bescobarm) [reply gemini/gemini.go:23]: i don't see it solved

File: telegram/commands/gemini/gemini.go (hunk 5 above)
Comment (CorreaJose13) [reply telegram/commands/gemini/gemini.go]: solved

File: gemini/gemini.go (hunk 6 above)
Comment (CorreaJose13) [reply gemini/gemini.go:23]: besides black you're blind

**Review by bescobarm — APPROVED** [review]

File: telegram/commands/gemini/gemini.go [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+	"errors"
+	"fmt"
+)
+
+var (
+	errNoPrompt = errors.New("En serio metiste un prompt vacío bobo hijueputa? Haz algo bien sub-humano 🙄")
+	forceText   = "responde solo texto"
+)
+
+func HandleGemini(bot *shared.Bot, chatID int64, userName, content string) error {
```
Comment (bescobarm) [inline telegram/commands/gemini/gemini.go:15]: userName is not used, remove it

File: telegram/commands/gemini/gemini.go [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+import (
+	"activity-tracker/gemini"
+	"activity-tracker/shared"
+	"errors"
+	"fmt"
+)
+
+var (
+	errNoPrompt = errors.New("En serio metiste un prompt vacío bobo hijueputa? Haz algo bien sub-humano 🙄")
+	forceText   = "responde solo texto"
```
Comment (bescobarm) [inline telegram/commands/gemini/gemini.go:12]: just out of consistent, add "," at the beginning of the string value

File: telegram/commands/gemini/gemini.go (hunk 9 above)
Comment (CorreaJose13) [reply telegram/commands/gemini/gemini.go:15]: we need to refactor (again) the code cuz there's a lot of func who has userName as param an never use it

Comment (CorreaJose13) [reply telegram/commands/gemini/gemini.go:15]: solved

File: telegram/commands/gemini/gemini.go (hunk 10 above)
Comment (CorreaJose13) [reply telegram/commands/gemini/gemini.go:12]: solved

File: telegram/commands/gemini/gemini.go (hunk 9 above)
Comment (bescobarm) [reply telegram/commands/gemini/gemini.go:15]: oki

**Review by bescobarm — APPROVED** [review]

**Pull request merged**